modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -402,22 +446,35 @@ public CompletableFuture<?> onCreate(ConfigurationNotificationEvent<Distribution
         public CompletableFuture<?> onDelete(ConfigurationNotificationEvent<DistributionZoneView> ctx) {
             updateMetaStorageOnZoneDelete(ctx.oldValue().zoneId(), ctx.storageRevision());
 
+            zonesTimers.remove(ctx.oldValue().zoneId());
+
             return completedFuture(null);
         }
 
         @Override
         public CompletableFuture<?> onUpdate(ConfigurationNotificationEvent<DistributionZoneView> ctx) {
-            updateMetaStorageOnZoneUpdate(ctx.storageRevision());
+            int zoneId = ctx.newValue().zoneId();
+
+            updateMetaStorageOnZoneUpdate(ctx.storageRevision(), zoneId);
+
+            int oldScaleUp = ctx.oldValue().dataNodesAutoAdjustScaleUp();
 
-            //TODO: Also add here rescheduling for the existing timers https://issues.apache.org/jira/browse/IGNITE-18121
+            int newScaleUp = ctx.newValue().dataNodesAutoAdjustScaleUp();
+
+            if (newScaleUp != Integer.MAX_VALUE && oldScaleUp != newScaleUp) {
+                // It is safe to zonesTimers.get(zoneId) in term of NPE because meta storage notifications are one-threaded
+                zonesTimers.get(zoneId).rescheduleOngoingScaleUp(newScaleUp);
```

Comment:
Why do you expect mandatory ongoing timer at this point? Let's say that that timer was already fired, should we consider it as an ongoing one, probably not.